File: fixtures/scale/ruby/large/lib/check_lattice.rb
require_relative 'lattice_encrypt'
require_relative 'lattice_decrypt'
require_relative 'corpus_rotate'

def check_lattice
  corpus_rotate.each do |text|
    sealed = lattice_encrypt(text, 6)
    opened = lattice_decrypt(sealed, 6)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_lattice.py
from cipherlab.ciphers.lattice import lattice_encrypt, lattice_decrypt
from cipherlab.specs.lattice_spec import lattice_spec
from cipherlab.corpus.corpus_rotate import corpus_rotate


def check_lattice():
    spec = lattice_spec()
    for text in corpus_rotate():
        sealed = lattice_encrypt(text, spec["key"])
        opened = lattice_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "rotate"
